From 51808ebe1f9f348fc2691647d137f20fd5f36c65 Mon Sep 17 00:00:00 2001
From: Robin <zhaoruibin@aliyun.com>
Date: Mon, 26 Jan 2026 11:50:39 +0800
Subject: [PATCH] fix: error getting package version number

make image  PACKAGES="app1=1.10 app2=2.0 app3"
get error version 2.0 for app3

Signed-off-by: Robin <zhaoruibin@aliyun.com>
---
 target/imagebuilder/files/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/target/imagebuilder/files/Makefile b/target/imagebuilder/files/Makefile
index 2b3c0e6ae0671f..d0348e51442270 100644
--- a/target/imagebuilder/files/Makefile
+++ b/target/imagebuilder/files/Makefile
@@ -166,6 +166,7 @@ endef
 define FormatPackages
 $(strip $(foreach pkg,$(strip $(subst ",,$(1))),
   $(eval pkg_name:=$(firstword $(subst =, ,$(pkg))))
+  $(eval pkg_ver:=)
   $(if $(findstring =,$(pkg)),$(eval pkg_ver:==$(lastword $(subst =, ,$(pkg)))))
   $(pkg_name)$(call GetABISuffix,$(pkg_name))$(pkg_ver)
 ))
 